stellargraph/data/explorer.py

```diff
@@ -780,3 +783,166 @@ def _check_neighbourhood_sizes(self, in_size, out_size):
             self._raise_error(
                 "The number of hops for the in and out neighbourhoods must be the same."
             )
+
+
+@experimental(reason="requires more thorough testing and documentation")
+class TemporalRandomWalk(GraphWalk):
+    """
+    Performs uniform temporal random walks on the given graph
+    """
+
+    def run(
+        self,
+        num_cw,
+        cw_size,
+        max_walk_length=80,
+        initial_edge_bias=None,
+        walk_bias=None,
+        p_walk_success_threshold=0.01,
+        seed=None,
+    ):
+        """
+        Perform a time respecting random walk starting from the root nodes.
+
+        Args:
+            num_cw (int): Total number of context windows to generate. For comparable
+                results to most other random walks, this should be a multiple of the number
+                of nodes in the graph.
+            cw_size (int): Size of context window. Also used as the minimum walk length,
+                since a walk must generate at least 1 context window for it to be useful.
+            max_walk_length (int): Maximum length of each random walk. Should be greater
+                than or equal to the context window size.
+            initial_edge_bias (str, optional): distribution to use when choosing a random
+                initial temporal edge to start from. Available options are:
+                * None (default) - the initial edge is picked from a uniform distribution
+                * "exponential" - heavily biased towards more recent edges.
+            walk_bias (str, optional): distribution to use when choosing a random
+                neighbour to walk through. Available options are:
+                * None (default) - neighbours are picked from a uniform distribution
+                * "exponential" - exponentially decaying probability, resulting in a bias
+                    towards shorter time gaps
+            p_walk_success_threshold (float): Lower bound for the proportion of successful
```

Comment:
The purpose of this parameter is not clear. If you are trying to avoid infinite loops (I suspect) because you cannot generate enough data for a given graph and random walk settings, then we need to explain more clearly how this parameter controls this behaviour.

Alternatively, we can display a progress bar (since we know how much data we need to generate, calculating the percent completed should be straightforward) and let the user decide if they want to kill the process if the progress bar is not moving. We can indicate this 'infinite loop' behaviour can happen in the class docstring.

Reply:
I've added a bit more description